packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -31,39 +39,92 @@ export abstract class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
-          const fileErrors = deployErrorParser.parse(stdErr);
+      try {
+        const deployParser = new ForceDeployResultParser(stdOut);
+        const errors = deployParser.getErrors();
+        if (errors) {
           handleDiagnosticErrors(
-            fileErrors,
+            errors,
             workspacePath,
             execFilePathOrPaths,
             ForceSourceDeployExecutor.errorCollection
           );
-        } catch (e) {
-          telemetryService.sendError(
-            'Error while creating diagnostics for vscode problem view.'
-          );
-          console.error(
-            'Error while creating diagnostics for vscode problem view.'
-          );
+        } else {
+          ForceSourceDeployExecutor.errorCollection.clear();
         }
-      } else {
-        ForceSourceDeployExecutor.errorCollection.clear();
+        this.outputResult(deployParser);
+      } catch (e) {
+        telemetryService.sendError(
+          'Error while creating diagnostics for vscode problem view.'
+        );
+        console.error(
+          'Error while creating diagnostics for vscode problem view.'
+        );
       }
     });
 
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private outputResult(parser: ForceDeployResultParser) {
+    const table = new Table();
+    const errors = parser.getErrors();
+    const successes = parser.getSuccesses();
+    const deployedSource = successes
+      ? successes.result.deployedSource
+      : undefined;
+
+    if (deployedSource) {
+      const outputTable = table.createTable(
+        (deployedSource as unknown) as Row[],
+        [
+          { key: 'state', label: nls.localize('table_header_state') },
+          { key: 'fullName', label: nls.localize('table_header_full_name') },
+          { key: 'type', label: nls.localize('table_header_type') },
+          { key: 'filePath', label: nls.localize('table_header_project_path') }
+        ]
+      );
+      channelService.appendLine('=== Deployed Source');
+      channelService.appendLine(outputTable);
+    }
+
+    if (errors) {
+      const { name, message, result } = errors;
+      if (result) {
+        const outputTable = table.createTable(
+          (errors.result as unknown) as Row[],
+          [
+            {
+              key: 'filePath',
+              label: nls.localize('table_header_project_path')
+            },
+            { key: 'error', label: nls.localize('table_header_errors') }
+          ]
+        );
+        channelService.appendLine('=== Deploy Errors');
+        channelService.appendLine(outputTable);
+      } else if (name && message) {
+        channelService.appendLine(`${name}: ${message}\n`);
```

Comment:
If there aren't any results in the error, at least show the name and message of the error to show what happened.